app/src/lib/utils/string.ts

```diff
@@ -11,3 +11,23 @@ export function hashCode(s: string) {
 	}
 	return hash.toString();
 }
+
+/**
+ * Checks if a string is a single character.
+ *
+ * @param char - The string to check.
+ * @returns `true` if the string is a single character, `false` otherwise.
+ */
+export function isChar(char: string) {
+	return char.length === 1;
+}
+
+/**
+ * Checks if a value is a string.
+ *
+ * @param s - The value to check.
+ * @returns A boolean indicating whether the value is a string.
+ */
+export function isStr(s: unknown): s is string {
+	return typeof s === 'string';
```

Comment:
Please use `==`

Reply:
Will do in a follow-up commit, but would you be please be open to reconsider this position at a later date?

Not only there are some cases in which even using TypeScript (e.g. when using union types) you need to use strict equality, but it's also just [faster than loose equality](https://stackoverflow.com/questions/12374815/javascript-vs-operators-performance).

Given, the performance gains are not huge but they are not 0. And I'd say that that in combination with the cases in which is still needed in TS it's something that should win against cross-language stylistic choices.

Ok that's what I needed to say haha 😅

Reply:
Hey! I'm interested, could you share an example of how a union can be problematic?

Reply:
Gladly!
The simplest example would be a something like this:
```ts
let example: string | number[];

function initializeExample() {
  example = [1];
}

initializeExample();

if (example == "1") {
  console.log("unsafe");
} else {
  console.log("safe");
}

// Console will log: unsafe
```
The TS linter would not catch this because the value is being set outside of the immediate scope in which the `if` statement is being evaluated, so it will still treat the union type as `string | number[]` instead of `number[]`.

Another example is with `any` which just completely disables TypeScript checks.
```ts
let example: any;

function initializeExample() {
  example = [1,2];
}

initializeExample();

if (example == "1,2") {
  console.log("unsafe");
} else {
  console.log("safe");
}

// Console will log: unsafe
```

Reply:
Hi @estib-vega,

We've talked about it internally, and we'll be moving over to `===`, so feel free to revert the commit, or I'll be coming in later this week to convert the project.